### ISSUE TITLE
FP8 quantization operations not functioning correctly in deeptools

### ISSUE THREAD

**Issue by Akash-Nayak** [torch-spyre/torch-spyre#2186]
## Description

FP8 quantization/casting workflows in deeptools do not appear to function correctly based on my experiments, which is currently blocking FP8 support validation and related testing.

During testing, multiple FP8 quantization paths produced unexpected behavior, including runtime failures and incorrect outputs.

---

### Affected Operations
- `qfp8` (activation quantization, single pad)
- `qfp8wt` (weight quantization, no pad)
- `qfp8mb` (minibatch activation quantization, single pad)

### Error
```
DtException: Fixed layout with too many dimensions
```

### Root Cause
All three operations create **KERNEL tensor type** for output instead of **OUTPUT tensor type**:
- KERNEL tensors require 6D layout: `[output_stick_dim, stick_dim, outer0, outer1, outer2, outer3]`
- torch-spyre generates 2D/3D layouts: `[mb, out]` or `[batch, seq, hidden]`
- DDL templates (quantization_single_pad.ddl, quantization_no_pad.ddl) expect 6D KERNEL layout

### Evidence (SDSC Analysis)
```json
{
  "Tensor1": {
    "tensorType": "KERNEL",  
    "dataFormat": "SEN143_FP8",
    "layoutDimOrder": ["mb", "out"],  
    "layoutDimSizes": [2, 64]
  }
}
```

**Expected for activations:**
```json
{
  "Tensor1": {
    "tensorType": "OUTPUT",  
    "dataFormat": "SEN143_FP8",
    "layoutDimOrder": ["mb", "out"],
    "layoutDimSizes": [2, 64]
  }
}
```

### Reproduction
**Branch Required**: https://github.com/Akash-Nayak/torch-spyre/tree/fp8-quantization-support

**Setup**:
```bash
# Clone and checkout the FP8 support branch
git clone https://github.com/Akash-Nayak/torch-spyre.git
cd torch-spyre
git checkout fp8-quantization-support

# Build torch-spyre
uv pip install -e . --no-deps --no-build-isolation
```

**Script**:
```python
#!/usr/bin/env python3
"""Reproduce qfp8 DDL layout error"""
import torch
import torch_spyre

print("Testing qfp8 operation (should fail with DDL error)\n")

def quantize_fp8(x):
    return torch.ops.spyre.qfp8(x)

compiled_fn = torch.compile(quantize_fp8, backend='inductor')

# Create test data
x_fp16 = torch.randn(2, 1, 4, dtype=torch.float16, device='spyre')

try:
    result = compiled_fn(x_fp16)
    print("❌ UNEXPECTED: Operation succeeded")
    print(f"Result: {result.cpu()}")
except Exception as e:
    print("✅ EXPECTED ERROR:")
    print(f"Error type: {type(e).__name__}")
    print(f"Error message: {str(e)}")
    if "Fixed layout with too many dimensions" in str(e):
        print("\n✅ DDL layout error confirmed")
    else:
        print("\n⚠️  Different error than expected")
```

**Expected Output:**
```
✅ EXPECTED ERROR:
Error type: DtException
Error message: Fixed layout with too many dimensions
✅ DDL layout error confirmed
```

### Impact
- Cannot quantize activations to FP8 using the operations qfp8/qfp8wt

---

## Issue 2: qfp8ch - "Anywhere Valid" Coordinate Mapping Bug

**See Issue #2185** for detailed "anywhere valid" coordinate arrangement analysis.

### Affected Operation
- `qfp8ch` (channel activation quantization, double pad)

### Error
**Data corruption for hidden_dim > 8: First 8 elements OK, rest become zeros**

### Additional Constraints (qfp8ch)
- **batch_size ≥ 2** (batch_size=1 causes AttributeError)
- **seq_len ≤ 16** (larger sequences cause AttributeError)
- **hidden_dim ≤ 8** (larger dimensions cause corruption)

---

### Affected Operation
- `qfp8chil` (interleaved activation quantization, double pad)

### Error
**All inputs produce constant output: -0.3125**

### Reproduction Script
```python
#!/usr/bin/env python3
"""Reproduce qfp8chil constant output bug"""
import torch
import torch_spyre

print("Testing qfp8chil constant output bug\n")

def quantize_fp8(x):
    return torch.ops.spyre.qfp8chil(x)

compiled_fn = torch.compile(quantize_fp8, backend='inductor')

# Test 1: Random positive values
print("Test 1: Random positive values")
x1 = torch.tensor([[1.033, 0.053, 1.498, 1.197]], dtype=torch.float16, device='spyre')
result1 = compiled_fn(x1).cpu()
print(f"Input:  {x1.cpu()[0]}")
print(f"Output: {result1[0]}")

# Test 2: Random negative values
print("\nTest 2: Random negative values")
x2 = torch.tensor([[-2.5, -1.3, -0.8, -3.1]], dtype=torch.float16, device='spyre')
result2 = compiled_fn(x2).cpu()
print(f"Input:  {x2.cpu()[0]}")
print(f"Output: {result2[0]}")

# Test 3: Large values
print("\nTest 3: Large values")
x3 = torch.tensor([[100.0, 200.0, 300.0, 400.0]], dtype=torch.float16, device='spyre')
result3 = compiled_fn(x3).cpu()
print(f"Input:  {x3.cpu()[0]}")
print(f"Output: {result3[0]}")
```

**Expected Output:**
```
Test 1: Random positive values
Input:  tensor([1.0332, 0.0532, 1.4980, 1.1973], dtype=torch.float16)
Output: tensor([-0.3125, -0.3125, -0.3125, -0.3125], dtype=torch.float8_e4m3fn)

Test 2: Random negative values
Input:  tensor([-2.5000, -1.3008, -0.7998, -3.0996], dtype=torch.float16)
Output: tensor([-0.3125, -0.3125, -0.3125, -0.3125], dtype=torch.float8_e4m3fn)

Test 3: Large values
Input:  tensor([100., 200., 300., 400.], dtype=torch.float16)
Output: tensor([-0.3125, -0.3125, -0.3125, -0.3125], dtype=torch.float8_e4m3fn)
```

### Impact
- qfp8chil unusable despite compiling successfully
- Cannot use as workaround for qfp8/qfp8wt/qfp8mb
- Blocks FP8 quantization for hidden_dim > 8

---

## Summary Table

| Operation | Issue | Impact |
|-----------|-------|--------|
| qfp8 | DDL layout error (KERNEL vs OUTPUT) | Cannot quantize activations |
| qfp8wt | DDL layout error (KERNEL vs OUTPUT) | Cannot quantize weights |
| qfp8mb | DDL layout error (KERNEL vs OUTPUT) | Cannot quantize minibatch |
| qfp8ch | Coordinate mapping (hidden_dim > 8) | 50% data loss |
| qfp8chil | Constant output (-0.3125) | Unusable for any input |

---

## Priority

**BLOCKER** - Blocks FP8 quantization support 

---

## Branch

https://github.com/Akash-Nayak/torch-spyre/tree/fp8-quantization-support

**Comment by Akash-Nayak** [comment]
FP8 operators `quantize_fp8_with_scale` and `dequantize_fp8_with_scale` for quantization and dequantization  were added in https://github.com/torch-spyre/torch-spyre/pull/2401. FP8 quantization uses the `qfp8ch` backend operation for activation quantization and FP8 dequantization uses `fp8todl16` backend operation. https://github.com/torch-spyre/torch-spyre/pull/2286 adds `quantize_weight_fp8_with_scale` for weight quantization that uses `qfp8wt` backend operation. 

```python
x_fp8 = torch.ops.spyre.quantize_fp8_with_scale(x, scale)
x_fp16 = torch.ops.spyre.dequantize_fp8_with_scale(x_fp8, scale)
```

With roundtrip testing (FP16 -> FP8 -> FP16) through `tests/inductor/test_fp8_operations.py`, it has been verified that data suffering from [anywhere valid issue](https://github.com/torch-spyre/torch-spyre/issues/2185) with `qfp8ch` is not corrupted on the Spyre device. 


Since, we have enabled FP8 quantization using `qfp8ch` and `qfp8wt`, and the `qfp8`, `qfp8mb`, and `qfp8chil` ops are not being used for the quantization, so I am closing this issue.

Related issues- 
https://github.com/torch-spyre/torch-spyre/issues/2492 
https://github.com/torch-spyre/torch-spyre/issues/2493
https://github.com/torch-spyre/torch-spyre/issues/1956
